tools/tcptop.py

```diff
@@ -223,60 +240,60 @@ def pid_to_comm(pid):
         with open(loadavg) as stats:
             print("%-8s loadavg: %s" % (strftime("%H:%M:%S"), stats.read()))
 
-    # IPv4:  build dict of all seen keys
-    keys = ipv4_recv_bytes
-    for k, v in ipv4_send_bytes.items():
-        if k not in keys:
-            keys[k] = v
+    ipv4_send_bytes_dict = ipv4_table_to_dict(ipv4_send_bytes)
+    ipv4_recv_bytes_dict = ipv4_table_to_dict(ipv4_recv_bytes)
+
+    # IPv4: build dict of all seen keys
+    keys = ipv4_send_bytes_dict.copy()
```

Comment:
Maybe the `dict` can be from `TCPSessionKey` keys to a pair of send / received bytes, and we just go through both send the received table to update the values? You can use Python's `defaultdict` to initialize the values to 0 if the key is not seen before.